include/xsf/mathieu/LICENSE

```diff
@@ -0,0 +1,21 @@
+MIT License
```

Comment:
Is there a reason why you're adding a new license file, rather than contributing this under `xsf`'s license? The latter is normally expected and would be a lot easier (license files propagate, even beyond this repo).

Reply:
I was going to comment that if @brorson has specific reasons to prefer an MIT license for this code, an entry will need to be added to https://github.com/scipy/xsf/blob/main/LICENSES_bundled.txt. It would of course be simpler to contribute this under `xsf`'s license though